src/init.cpp

```diff
@@ -471,6 +471,7 @@ void SetupServerArgs(ArgsManager& argsman)
     argsman.AddArg("-onion=<ip:port>", "Use separate SOCKS5 proxy to reach peers via Tor onion services, set -noonion to disable (default: -proxy)", ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
     argsman.AddArg("-i2psam=<ip:port>", "I2P SAM proxy to reach I2P peers and accept I2P connections (default: none)", ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
     argsman.AddArg("-i2pacceptincoming", "If set and -i2psam is also set then incoming I2P connections are accepted via the SAM proxy. If this is not set but -i2psam is set then only outgoing connections will be made to the I2P network. Ignored if -i2psam is not set. Listening for incoming I2P connections is done through the SAM proxy, not by binding to a local address and port (default: 1)", ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
+    argsman.AddArg("-i2ptransientout", strprintf("Generate a transient, one-time I2P address for ourselves when making I2P outbound connection and discard it once the connection is closed. Does not affect the address for accepting incoming I2P connections which is persistent. Ignored if -i2psam is not set. (default: %u)", DEFAULT_I2P_TRANSIENT_OUT), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
```

Comment:
Took some of this. "use" is better than "announce" here because we do not actually announce our I2P address anywhere (for outgoing connections). The peer has knowledge of it from the underlying networking layer.